### PR TITLE
Remove Support for Injecting ILogger Directly

### PR DESCRIPTION
**Pull request by barronpm**
**Changes**
Remove the functionality for injecting ILogger directly, replaces all usages with generic versions or `LoggerFactory`

### CODE REVIEW

**Comment by barronpm** [comment]
The goal of this PR is not to remove all instances of non-generic `ILogger`, but rather to remove support for it in our dependency injection.

**Comment by crobibero** [comment]
> 
> 
> The goal of this PR is not to remove all instances of non-generic `ILogger`, but rather to remove support for it in our dependency injection.

Sorry, I must have misunderstood what the following meant.
> replaces all usages with generic versions or LoggerFactory

**Comment by barronpm** [comment]
Yeah, I probably could have phrased that better

File: Emby.Server.Implementations/Library/Resolvers/Audio/MusicArtistResolver.cs [hunk 1]
```diff
@@ -15,20 +15,20 @@ namespace Emby.Server.Implementations.Library.Resolvers.Audio
     /// </summary>
     public class MusicArtistResolver : ItemResolver<MusicArtist>
     {
-        private readonly ILogger _logger;
+        private readonly ILogger<MusicAlbumResolver> _logger;
```
Comment (dkanada) [inline Emby.Server.Implementations/Library/Resolvers/Audio/MusicArtistResolver.cs:18]: The class is MusicArtist so why use the Album one?

Comment (barronpm) [reply Emby.Server.Implementations/Library/Resolvers/Audio/MusicArtistResolver.cs:18]: That instance is passed through to created instances of `MusicAlbumResolver` and never used by the `MusicArtistResolver` class itself

**Comment by Bond-009** [comment]
And now all plugins are broken... (educated guess ;) )

**Comment by JustAMan** [comment]
They do it frequently, so it's no biggie xD

**Comment by mark-monteiro** [comment]
Can we do something to flag PRs like this? Maybe a new label `breaking change`?

**Comment by dkanada** [comment]
They happen frequently enough that it became useless to label them, but I mean we can always enable it again and have like half our PRs fail the checks. Actually, SonarCloud is already causing half the PRs to fail the checks so it would probably inch up to 70% or so.

**Comment by dkanada** [comment]
I stand corrected, only about 10% are failing right now. The former point still stands though, we break the C# ABI so often it fails a lot of the PRs and there's currently know way to know which plugins will be affected by a given ABI change.

**Comment by PrplHaz4** [comment]
> 
> 
> I stand corrected, only about 10% are failing right now. The former point still stands though, we break the C# ABI so often it fails a lot of the PRs and there's currently know way to know which plugins will be affected by a given ABI change.

It doesn't necessarily need to be automated - if there's a known high-impact break it should be called out somehow, no?

**Comment by JustAMan** [comment]
Plugins don't work in nightlies so far anyway, and we'll fix them near the release, so I don't see a point arguing here.

**Comment by MrTimscampi** [comment]
I'm still getting the following with no plugin installed (aside from the mandatory ones: musicbrainz and audiodb):
```[2020-05-14 21:55:40.050 +02:00] [ERR] [16] Emby.Server.Implementations.HttpServer.HttpListenerHost: Error processing request. URL: ""
System.InvalidOperationException: Unable to resolve service for type 'Microsoft.Extensions.Logging.ILogger' while attempting to activate 'MediaBrowser.Api.Playback.MediaInfoService'.
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.ConstructorMatcher.CreateInstance(IServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance(IServiceProvider provider, Type instanceType, Object[] parameters)
   at Emby.Server.Implementations.Services.ServiceController.Execute(HttpListenerHost httpHost, Object requestDto, IRequest req) in /home/mrtimscampi/.cache/yay/jellyfin-git/src/jellyfin/Emby.Server.Implementations/Services/ServiceController.cs:line 184
   at Emby.Server.Implementations.Services.ServiceHandler.ProcessRequestAsync(HttpListenerHost httpHost, IRequest httpReq, HttpResponse httpRes, ILogger logger, CancellationToken cancellationToken) in /home/mrtimscampi/.cache/yay/jellyfin-git/src/jellyfin/Emby.Server.Implementations/Services/ServiceHandler.cs:line 80
   at Emby.Server.Implementations.HttpServer.HttpListenerHost.RequestHandler(IHttpRequest httpReq, String urlString, String host, String localPath, CancellationToken cancellationToken) in /home/mrtimscampi/.cache/yay/jellyfin-git/src/jellyfin/Emby.Server.Implementations/HttpServer/HttpListenerHost.cs:line 540
```

**Pull request merged**